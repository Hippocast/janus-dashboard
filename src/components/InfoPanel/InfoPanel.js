import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';

import './InfoPanel.css';

const propTypes = {
    icon: PropTypes.node,
    text: PropTypes.string.isRequired,
};

const b = block('info-panel');

const InfoPanel = ({ icon, text }) => {
    return (
        <div className={b.mix('j-pane')()}>
            {
                icon && icon
            }
            <p className={b('text')()}>{ text }</p>
        </div>
    );
};

InfoPanel.propTypes = propTypes;

export default InfoPanel;
