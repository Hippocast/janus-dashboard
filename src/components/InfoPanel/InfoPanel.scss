@import 'variables.scss';

.info-panel {
    display: flex;
    flex-direction: column;
    align-items: center;

    &__text {
        font-weight: $hf-font-weight;
        line-height: 1.5;
    }
}
