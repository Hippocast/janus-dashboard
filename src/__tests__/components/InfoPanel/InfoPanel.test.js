import React from 'react';
import renderer from 'react-test-renderer';

import InfoPanel from '../../../components/InfoPanel/InfoPanel';

describe('InfoPanel component', () => {
    const requiredProps = {
        text: 'mock-text',
    };

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <InfoPanel
                    {...requiredProps}
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders icon if passed', () => {
        const passedProps = {
            icon: 'mock-node',
        };
        const tree = renderer
            .create(
                <InfoPanel
                    {...requiredProps}
                    {...passedProps}
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
