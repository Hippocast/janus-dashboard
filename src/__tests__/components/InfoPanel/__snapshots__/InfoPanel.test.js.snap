// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel component renders correctly 1`] = `
<div
  className="info-panel j-pane"
>
  <p
    className="info-panel__text"
  >
    mock-text
  </p>
</div>
`;

exports[`InfoPanel component renders icon if passed 1`] = `
<div
  className="info-panel j-pane"
>
  mock-node
  <p
    className="info-panel__text"
  >
    mock-text
  </p>
</div>
`;
